src/d_iwad.c

```diff
@@ -663,6 +663,14 @@ static void AddXdgDirs(void)
     // ~/.local/share/games/doom) as a user-writeable extension to
     // the usual /usr/share/games/doom location.
     AddIWADDir(M_StringJoin(env, "/games/doom", NULL));
+    #ifdef __HAIKU__
+    AddIWADDir(M_StringJoin(env, "/doom", NULL));
+    AddIWADDir(M_StringJoin(env, "/doomdata", NULL));
+    AddIWADDir(M_StringJoin(env, "/chocolate-doom", NULL));
+    env = getenv("HOME");
```

Comment:
I would prefer this in its own variable. It will clobber the previous `env`. While it appears to be correct for now if someone is careless when adding more stuff here it might lead to non-obvious bugs.